src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/DoNotIgnoreMethodResults.cs

```diff
@@ -140,6 +150,13 @@ public override void Initialize(AnalysisContext analysisContext)
                 INamedTypeSymbol? xunitAssertType = compilationContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.XunitAssert);
                 INamedTypeSymbol? linqEnumerableType = compilationContext.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemLinqEnumerable);
 
+                if (!(compilationContext.Compilation.SyntaxTrees.FirstOrDefault() is SyntaxTree tree))
```

Comment:
This approach should only be used if it is not possible to associate a specific syntax tree with the place where we need the option. In this case, that is not true as every operation block action has an associated syntax tree.